### ISSUE TITLE
Straighten out virtualenv configuration

### ISSUE THREAD

**Issue by conorsch** [freedomofpress/django_stack#11]
The role using Ansible exception handling to configure the virtualenv according to two different strategies:

1. using system-wide python
2. using virtualenv python

In practice this generally means that a run of the role emits a gigantic block of terrifying crimson text, then blithely continues to execute by calling an identically named task (`TASK [freedomofpress.django-stack : Setup virtualenv from requirements file] ***`) that completes successfully.

Due to the exception handling, there's not even an `...ignoring` message on the "failed" task, as would be the case with the `ignore_errors: true` parameter.

Can't we just assume that the virtualenv python is the version we should be using?

**Comment by msheiny** [comment]
Yeah to be honest.... this is where better documentation on my part would have been handy. I can't remember exactly why I added this but I kept running into a problem and I was banging my head against the wall to try and solve it. I think the issue was that when the virtualenv folder isnt created yet the first pass strategy works but when the folder is already installed the second strategy works. I'm not really sure why this is the case but I kept running into it randomnly. I don't know what the long-term solution is but I think we might be able to at least swap strategy order -- that would speed things up since the majority of the time strategy 2 is needed. Does that make sense? I could have just done something really dumb or there is something I overlooked.

**Comment by conorsch** [comment]
>we might be able to at least swap strategy order -- that would speed things up since the majority of the time strategy 2 is needed.

:+1: That sounds like a great compromise. It'll Just Work™ with the majority of deployments that _should_ be using a virtualenv anyway, and for edge cases where that's not true, the failover task should handle nicely.

Having unique names for the tasks would go a long way in making the playbook output more intelligible.

**Comment by msheiny** [comment]
Uggghhh it doesn't work when swapped. This issue is really dumb and I'm convinced its broken behavior in ansible. There a couple of open issues related to `pip` and `virtualenv` -- none specifically covers this weird use-case as far as I could tell. I dunno, I started going down this rabbit hole before and got really frustrated -- maybe this is one of those @garrettr can fix in 5 minutes type things!? :smile:

**Comment by msheiny** [comment]
to clarify -- @conorsch  it doesnt work WHEN the virtualenv folder doesnt exist when you swap

i think what happens is that first run creates the `virtualenv` folder, it fails but then gets rescued and resolves ... maybe there is another virtualenv module outside of pip ...

it looks like the first run is also ignoring the python version I told it to utilize and using `pip2`

```
TASK [django_stack : VirtualEnv from requirements (new virtualenv folder)] *****
task path: /home/mike/Documents/Git/django_stack/tasks/django_service.yml:30
fatal: [djangoserver]: FAILED! => {"changed": false, "cmd": "/usr/bin/pip2 install -r /var/www/django/securethenews/requirements.txt", "failed": true, "msg": "stdout: Already using interpreter /usr/bin/python3\nUsing base prefix '/usr'\nNew python executable in /home/gcorn/securethenews/bin/python3\nAlso creating executable in /home/gcorn/securethenews/bin/python\nInstalling setuptools, pip...done.\nDownloading/unpacking Django>=1.10,<1.11 (from -r /var/www/django/securethenews/requirements.txt (line 1))\nDownloading/unpacking django-logging-json==1.5.3 (from -r /var/www/django/securethenews/requirements.txt (line 2))\n  Downloading django-logging-json-1.5.3.tar.gz\n  Running setup.py (path:/tmp/pip-build-NufS3I/django-logging-json/setup.py) egg_info for package django-logging-json\n    \nDownloading/unpacking gunicorn==19.6.0 (from -r /var/www/django/securethenews/requirements.txt (line 3))\nDownloading/unpacking Pillow==3.4.2 (from -r /var/www/django/securethenews/requirements.txt (line 4))\n  Running setup.py (path:/tmp/pip-build-NufS3I/Pillow/setup.py) egg_info for package Pillow\n    \n    warning: no files found matching '*.sh'\n    no previously-included directories found matching 'docs/_static'\n    warning: no previously-included files found matching '.coveragerc'\n    warning: no previously-included files found matching '.editorconfig'\n    warning: no previously-included files found matching '.landscape.yaml'\n    warning: no previously-included files found matching 'appveyor.yml'\n    warning: no previously-included files found matching 'build_children.sh'\n    warning: no previously-included files found matching 'tox.ini'\n    warning: no previously-included files matching '.git*' found anywhere in distribution\n    warning: no previously-included files matching '*.pyc' found anywhere in distribution\n    warning: no previously-included files matching '*.so' found anywhere in distribution\nDownloading/unpacking psycopg2==2.6.2 (from -r /var/www/django/securethenews/requirements.txt (line 5))\n  Running setup.py (path:/tmp/pip-build-NufS3I/psycopg2/setup.py) egg_info for package psycopg2\n    \nDownloading/unpacking wagtail>=1.7,<1.8 (from -r /var/www/django/securethenews/requirements.txt (line 6))\nDownloading/unpacking wagtailmenus>=1.6,<1.7 (from -r /var/www/django/securethenews/requirements.txt (line 7))\n  Running setup.py (path:/tmp/pip-build-NufS3I/wagtailmenus/setup.py) egg_info for package wagtailmenus\n    \n    warning: no files found matching '*.scss' under directory 'wagtailmenus/static'\n    no previously-included directories found matching 'wagtailmenus/development'\nRequirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python2.7/dist-packages (from django-logging-json==1.5.3->-r /var/www/django/securethenews/requirements.txt (line 2))\nDownloading/unpacking elasticsearch>=2.0.0,<3.0.0 (from django-logging-json==1.5.3->-r /var/www/django/securethenews/requirements.txt (line 2))\nDownloading/unpacking django-modelcluster>=2.0,<3.0 (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\n  Downloading django-modelcluster-2.0.tar.gz\n  Running setup.py (path:/tmp/pip-build-NufS3I/django-modelcluster/setup.py) egg_info for package django-modelcluster\n    \nDownloading/unpacking beautifulsoup4>=4.5.1 (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\nDownloading/unpacking Willow>=0.4,<0.5 (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\nDownloading/unpacking djangorestframework>=3.1.3 (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\nDownloading/unpacking django-taggit>=0.18,<0.19 (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\nRequirement already satisfied (use --upgrade to upgrade): html5lib>=0.999,<1 in /usr/lib/python2.7/dist-packages (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\nDownloading/unpacking django-treebeard>=3.0,<5.0 (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\n  Running setup.py (path:/tmp/pip-build-NufS3I/django-treebeard/setup.py) egg_info for package django-treebeard\n    \nDownloading/unpacking Unidecode>=0.04.14 (from wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\n  Running setup.py (path:/tmp/pip-build-NufS3I/Unidecode/setup.py) egg_info for package Unidecode\n    \nRequirement already satisfied (use --upgrade to upgrade): urllib3>=1.8,<2.0 in /usr/lib/python2.7/dist-packages (from elasticsearch>=2.0.0,<3.0.0->django-logging-json==1.5.3->-r /var/www/django/securethenews/requirements.txt (line 2))\nDownloading/unpacking pytz>=2015.2 (from django-modelcluster>=2.0,<3.0->wagtail>=1.7,<1.8->-r /var/www/django/securethenews/requirements.txt (line 6))\nInstalling collected packages: Django, django-logging-json, gunicorn, Pillow, psycopg2, wagtail, wagtailmenus, elasticsearch, django-modelcluster, beautifulsoup4, Willow, djangorestframework, django-taggit, django-treebeard, Unidecode, pytz\nCleaning up...\nException:\nTraceback (most recent call last):\n  File \"/usr/lib/python2.7/dist-packages/pip/basecommand.py\", line 122, in main\n    status = self.run(options, args)\n  File \"/usr/lib/python2.7/dist-packages/pip/commands/install.py\", line 295, in run\n    requirement_set.install(install_options, global_options, root=options.root_path)\n  File \"/usr/lib/python2.7/dist-packages/pip/req.py\", line 1436, in install\n    requirement.install(install_options, global_options, *args, **kwargs)\n  File \"/usr/lib/python2.7/dist-packages/pip/req.py\", line 672, in install\n    self.move_wheel_files(self.source_dir, root=root)\n  File \"/usr/lib/python2.7/dist-packages/pip/req.py\", line 902, in move_wheel_files\n    pycompile=self.pycompile,\n  File \"/usr/lib/python2.7/dist-packages/pip/wheel.py\", line 214, in move_wheel_files\n    clobber(source, lib_dir, True)\n  File \"/usr/lib/python2.7/dist-packages/pip/wheel.py\", line 204, in clobber\n    os.makedirs(destdir)\n  File \"/usr/lib/python2.7/os.py\", line 157, in makedirs\n    mkdir(name, mode)\nOSError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/django'\n\nStoring debug log for failure in /home/gcorn/.pip/pip.log\n"}
```

**Comment by msheiny** [comment]
Got fixed in PR #20
